Make onrejectionhandled and onunhandledrejection work correctly on body elements

annevk opened the pull request:
#### ad9a7f1bea6fefdab6ccff7691355e2e0ea73aa7
<pre>
Make onrejectionhandled and onunhandledrejection work correctly on body elements
<a href="https://bugs.webkit.org/show_bug.cgi?id=278907">https://bugs.webkit.org/show_bug.cgi?id=278907</a>
<a href="https://rdar.apple.com/135401362">rdar://135401362</a>

Reviewed by Darin Adler.

Presumably this was not done in 245224@main to keep that a
straightforward refactoring. However, as the risk is minimal we should
align with the HTML standard here.

Tests are also synchronized and changes are exported here:
<a href="https://github.com/web-platform-tests/wpt/pull/48017">https://github.com/web-platform-tests/wpt/pull/48017</a>

* LayoutTests/imported/w3c/web-platform-tests/html/webappapis/scripting/events/event-handler-all-global-events-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/html/webappapis/scripting/events/event-handler-all-global-events.html:
* LayoutTests/imported/w3c/web-platform-tests/html/webappapis/scripting/events/event-handler-attributes-body-window-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/html/webappapis/scripting/events/event-handler-attributes-windowless-body-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/html/webappapis/scripting/events/event-handler-spec-example-expected.txt: Removed.
* LayoutTests/imported/w3c/web-platform-tests/html/webappapis/scripting/events/invalid-uncompiled-raw-handler-compiled-late-expected.txt: Removed.
* LayoutTests/imported/w3c/web-platform-tests/html/webappapis/scripting/events/invalid-uncompiled-raw-handler-compiled-once-expected.txt: Removed.
* LayoutTests/imported/w3c/web-platform-tests/html/webappapis/scripting/events/messageevent-constructor.https-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/html/webappapis/scripting/events/messageevent-constructor.https.html:
* LayoutTests/imported/w3c/web-platform-tests/html/webappapis/scripting/events/resources/event-handler-body.js:
(eventHandlerTest):
* LayoutTests/imported/w3c/web-platform-tests/html/webappapis/scripting/events/w3c-import.log:
* LayoutTests/platform/mac-wk1/imported/w3c/web-platform-tests/html/webappapis/scripting/events/messageevent-constructor.https-expected.txt:
* Source/WebCore/html/HTMLBodyElement.cpp:
(WebCore::HTMLBodyElement::eventNameForWindowEventHandlerAttribute):

Canonical link: <a href="https://commits.webkit.org/283698@main">https://commits.webkit.org/283698@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/80050e6a6b7b99e0d73642fef45e5b4d64cffd4e

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/67099 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/46478 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/19731 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/71134 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/18232 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/69217 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/54276 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/18024 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/53820 "Passed tests") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/12272 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/70166 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/42753 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/58060 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/34358 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/39425 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/15454 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/16586 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/61387 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/15795 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/72837 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/11058 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/15149 "Passed tests") | [❌ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/61285 "Found 5 new test failures: fast/images/animated-jpegxl-loop-count.html fast/images/jpegxl-as-image.html fast/images/jpegxl-image-decoding.html fast/images/jpegxl-with-color-profile.html http/tests/images/repaint-garbled.html (failure)") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/11091 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/58119 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/61363 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/9093 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/2696 "Passed tests") | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/10177 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/42283 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/43360 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/44546 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/43101 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

darinadler commented on Source/WebCore/html/HTMLBodyElement.cpp:
```diff
@@ -120,9 +120,6 @@ const AtomString& HTMLBodyElement::eventNameForWindowEventHandlerAttribute(const
     static NeverDestroyed map = [] {
         EventHandlerNameMap map;
         JSHTMLBodyElement::forEachWindowEventHandlerContentAttribute([&] (const AtomString& attributeName, const AtomString& eventName) {
-            // FIXME: Remove these special cases. These have has an [WindowEventHandler] line in the IDL but were not in this map before, so this preserves behavior.
```
As the author of this FIXME, I want to think you for doing it!

darinadler commented on LayoutTests/imported/w3c/web-platform-tests/html/webappapis/scripting/events/event-handler-all-global-events-expected.txt:
```diff
@@ -58,7 +58,7 @@ FAIL oncontextlost: must be on the appropriate locations for GlobalEventHandlers
 FAIL oncontextlost: the content attribute must be compiled into a function as the corresponding property assert_equals: The oncontextlost property must be a function expected "function" but got "undefined"
 FAIL oncontextlost: the content attribute must execute when an event is dispatched assert_true: Dispatching an event must run the code expected true got undefined
-PASS oncontextlost: dispatching an Event at a <meta> element must trigger element.oncontextlost
+FAIL oncontextlost: dispatching an Event at a <meta> element must trigger element.oncontextlost assert_true: expected true got false
```
Is this OK?

annevk replied:
Yeah, this is a result of me making that test more strict. Before this change it would pass if the event listener didn't run.

darinadler commented on LayoutTests/imported/w3c/web-platform-tests/html/webappapis/scripting/events/event-handler-all-global-events-expected.txt:
```diff
@@ -68,7 +68,7 @@ FAIL oncontextrestored: must be on the appropriate locations for GlobalEventHand
 FAIL oncontextrestored: the content attribute must be compiled into a function as the corresponding property assert_equals: The oncontextrestored property must be a function expected "function" but got "undefined"
 FAIL oncontextrestored: the content attribute must execute when an event is dispatched assert_true: Dispatching an event must run the code expected true got undefined
-PASS oncontextrestored: dispatching an Event at a <meta> element must trigger element.oncontextrestored
+FAIL oncontextrestored: dispatching an Event at a <meta> element must trigger element.oncontextrestored assert_true: expected true got false
```
Is this OK?

annevk replied:
Same as above.

annevk committed:
Make onrejectionhandled and onunhandledrejection work correctly on body elements
https://bugs.webkit.org/show_bug.cgi?id=278907
rdar://135401362

Reviewed by Darin Adler.

Presumably this was not done in 245224@main to keep that a
straightforward refactoring. However, as the risk is minimal we should
align with the HTML standard here.

Tests are also synchronized and changes are exported here:
https://github.com/web-platform-tests/wpt/pull/48017

* LayoutTests/imported/w3c/web-platform-tests/html/webappapis/scripting/events/event-handler-all-global-events-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/html/webappapis/scripting/events/event-handler-all-global-events.html:
* LayoutTests/imported/w3c/web-platform-tests/html/webappapis/scripting/events/event-handler-attributes-body-window-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/html/webappapis/scripting/events/event-handler-attributes-windowless-body-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/html/webappapis/scripting/events/event-handler-spec-example-expected.txt: Removed.
* LayoutTests/imported/w3c/web-platform-tests/html/webappapis/scripting/events/invalid-uncompiled-raw-handler-compiled-late-expected.txt: Removed.
* LayoutTests/imported/w3c/web-platform-tests/html/webappapis/scripting/events/invalid-uncompiled-raw-handler-compiled-once-expected.txt: Removed.
* LayoutTests/imported/w3c/web-platform-tests/html/webappapis/scripting/events/messageevent-constructor.https-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/html/webappapis/scripting/events/messageevent-constructor.https.html:
* LayoutTests/imported/w3c/web-platform-tests/html/webappapis/scripting/events/resources/event-handler-body.js:
(eventHandlerTest):
* LayoutTests/imported/w3c/web-platform-tests/html/webappapis/scripting/events/w3c-import.log:
* LayoutTests/platform/mac-wk1/imported/w3c/web-platform-tests/html/webappapis/scripting/events/messageevent-constructor.https-expected.txt:
* Source/WebCore/html/HTMLBodyElement.cpp:
(WebCore::HTMLBodyElement::eventNameForWindowEventHandlerAttribute):

Canonical link: https://commits.webkit.org/283698@main
